File: game/Classic.java
package game;

import game.utilities.HangmanEngine;
import game.utilities.SecureInput;
import game.utilities.HangmanGraphicsEngine;

public class Classic {
    public static void runGame() {

        //Show Start Screen
        System.out.println("\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n");
        System.out.println("Classic Game:");

        //Initialize Engine
        HangmanEngine game = new HangmanEngine();

        //Start Game
        String guess = "";
        boolean gameOver = false;
        int maxGuesses = 6;
        do {

            HangmanGraphicsEngine.drawHangman(maxGuesses);
            game.printGameState();
            System.out.println("Remaining guesses: " + maxGuesses);

            System.out.print("Enter a letter: ");
            guess = SecureInput.secureStringChar();
            if(!game.processGuess(guess)) {
                maxGuesses--;
            }

            if (maxGuesses == 0) {
                gameOver = true;
            }
            if (game.isGuessed()) {
                gameOver = true;
            }

            System.out.println("\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n");

        } while(!gameOver);

        HangmanGraphicsEngine.drawHangman(maxGuesses);
        game.printPostGameState();

    }
}

File: game/classic.c
#include <stdlib.h>
#include <stdio.h>
#include <string.h>
#include "utilities/hangman_engine.h"
#include "utilities/hangman_graphics_engine.h"
#include "utilities/secure_input.h"

static void clear_terminal(){
	printf("\033[2J\033[2J");
}

void run_classic_game(){
	int game_over = 0;
	int max_guesses = 6;
	char *guess;

	clear_terminal();
	printf("Classic Game:\n");
	hangman_engine_init();

	do{
		draw_hangman(max_guesses);
		print_game_state();
		printf("Remaining guesses: %d\nEnter your guess: ", max_guesses);
		guess = secure_string();
		if(strlen(guess) == 1){
			if(!hangman_engine_process_guess_char(guess[0])){
				max_guesses--;
			}
			free(guess);
		} else {
			if(!hangman_engine_process_guess_string(guess)){
				max_guesses--;
			}
			free(guess);
		}

		game_over = max_guesses == 0 || !strcmp(hangman_engine_word, hangman_engine_guess_string);
		clear_terminal();
	} while(!game_over);

	draw_hangman(max_guesses);
	print_post_game_state();
	printf("\nEnter any key to return to menu: ");
	fgetc(stdin);
	hangman_engine_exit();
}

